### PR TITLE
Fix coverity issues by adding lacking CUDA_CALL

### PR DESCRIPTION
**Pull request by banasraf**
Signed-off-by: Rafal <Banas.Rafal97@gmail.com>

#### Why we need this PR?
- It fixes coverity issues for unchecked results of CUDA calls.

#### What happened in this PR?
 - What solution was applied:
     searched for the following calls: 
     cudaMemcpy, cudaMemcpyAsync, cudaMemset, cudaMalloc, cudaEventRecord, cudaEventElapsedTime, cudaEventSynchronize, cudaStreamSynchronize, cudaStreamWaitEvent, cudaDeviceSynchronize and added lacking CUDA_CALLs
 - Affected modules and functionalities:
     *multiple files*
 - Key points relevant for the review:
     * not much important stuff here*
 - Validation and testing:
     *NA*
 - Documentation (including examples):
     *NA*

**JIRA TASK**: *DALI-1965*

### CODE REVIEW

**Review by jantonguirao — APPROVED** [review]

**Review by JanuszL — APPROVED** [review]

**Comment by banasraf** [comment]
!build

**Review by JanuszL — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

**Review by JanuszL — APPROVED** [review]

File: dali/kernels/imgproc/convolution/cutlass/device/gemm.h [hunk 1]
```diff
@@ -469,8 +469,9 @@ class Conv {
     }
 
     size_t params_sizeof = host_params_.size() * sizeof(SampleParams);
-    cudaMemcpyAsync(params_.params, host_params_.data(), params_sizeof, cudaMemcpyHostToDevice,
-                    stream);
+    CUDA_CALL(
```
Comment (mzient) [inline dali/kernels/imgproc/convolution/cutlass/device/gemm.h:472]: This is actually wrong - the status is checked and the function is apparently not supposed to throw.

File: include/dali/core/per_stream_pool.h [hunk 2]
```diff
@@ -165,7 +165,7 @@ class PerStreamPool {
   friend class ObjectLease;
 
   void Release(std::unique_ptr<ListNode> node) {
-    cudaEventRecord(node->event, node->stream);
+    CUDA_CALL(cudaEventRecord(node->event, node->stream));
```
Comment (mzient) [inline include/dali/core/per_stream_pool.h:168]:
```suggestion
    CUDA_DTOR_CALL(cudaEventRecord(node->event, node->stream));
```

**Review by mzient — CHANGES_REQUESTED** [review]
Replace CUDA_CALL in a destructor with CUDA_DTOR CALL.
Remove CUDA_CALL from a function returning a status.

File: dali/operators/generic/constant.cu [hunk 3]
```diff
@@ -66,7 +66,8 @@ void FillTensorList(
 
     int n = tmp.size() * sizeof(Dst);
     int N = shape.num_samples();
-    cudaMemcpyAsync(dst.mutable_tensor<Dst>(0), tmp.data(), n, cudaMemcpyHostToDevice, stream);
+    CUDA_CALL(
+      cudaMemcpyAsync(dst.mutable_tensor<Dst>(0), tmp.data(), n, cudaMemcpyHostToDevice, stream));
     int copied = 1;
     // this loop doubles the data in GPU memory, so that there are log2(N) memcpys at most
     while (copied < N) {
```
Comment (mzient) [inline dali/operators/generic/constant.cu:73]: missing CUDA_CALL in line 76

File: dali/operators/generic/constant.cu [hunk 4]
```diff
@@ -66,7 +66,8 @@ void FillTensorList(
 
     int n = tmp.size() * sizeof(Dst);
     int N = shape.num_samples();
-    cudaMemcpyAsync(dst.mutable_tensor<Dst>(0), tmp.data(), n, cudaMemcpyHostToDevice, stream);
+    CUDA_CALL(
```
Comment (mzient) [inline dali/operators/generic/constant.cu:69]: The part that calls a kernel isn't guarded.
Perhaps there should be a `CUDA_CALL(cudaGetLastError())` at call site instead, to capture all possible errors?

File: dali/kernels/reduce/reduce_all_gpu_test.cu [hunk 5]
```diff
@@ -138,32 +140,37 @@ void ReduceAllGPUTest<Reduction>::TestReduceBatched() {
...
   // sample sizes
-  cudaMemcpy(gpu_sizes.get(), sizes.data(), samples * sizeof(*gpu_sizes), cudaMemcpyHostToDevice);
+  CUDA_CALL(
+    cudaMemcpy(gpu_sizes.get(), sizes.data(), samples * sizeof(*gpu_sizes),
+               cudaMemcpyHostToDevice));
 
   // warm-up
   ReduceAllBatchedKernel<float><<<1, block>>>(
       out_data.get(), gpu_dev_ptrs.get(), gpu_sizes.get(), R);
   cudaDeviceSynchronize();
```
[the remark this answers is not in the document]
Comment (banasraf) [reply dali/kernels/reduce/reduce_all_gpu_test.cu]: done

File: dali/kernels/imgproc/convolution/cutlass/device/gemm.h (hunk 1 above)
Comment (banasraf) [reply dali/kernels/imgproc/convolution/cutlass/device/gemm.h]: done

File: dali/operators/generic/constant.cu (hunk 3 above)
Comment (banasraf) [reply dali/operators/generic/constant.cu:73]: done

**Review by mzient — APPROVED** [review]

**Comment by banasraf** [comment]
!build

**Pull request merged**